src/meta/app_env_validator.cpp

```diff
@@ -46,6 +46,27 @@ bool check_slow_query(const std::string &env_value, std::string &hint_message)
     return true;
 }
 
+bool check_deny_client(const std::string &env_value, std::string &hint_message)
+{
+    std::vector<std::string> sub_sargs;
+    utils::split_args(env_value.c_str(), sub_sargs, '*', true);
+
+    std::string invalid_hint_message = "Invalid deny client args, valid include: reject*all, "
+                                       "reject*write, reject*read; delay*all, delay*write, "
+                                       "delay*read";
+    if (sub_sargs.size() != 2) {
+        hint_message = invalid_hint_message;
+        return false;
+    }
+    if (sub_sargs[0].empty() || sub_sargs[1].empty() ||
```

Comment:
`sub_sargs[0].empty() || sub_sargs[1].empty() ||` is not necessary while you have checked their values.